src/msg/async/dpdk/dpdk_rte.cc

```diff
@@ -137,17 +137,21 @@ namespace dpdk {
       if (!rte_initialized) {
         /* initialise the EAL for all */
         int ret = rte_eal_init(cargs.size(), cargs.data());
-        if (ret < 0)
-          return;
+	if (ret < 0) {
+          std::unique_lock<std::mutex> locker(lock);
```

Comment:
no need to specify the template argument, as, C++17 is able to deduce the template argument type. see https://en.cppreference.com/w/cpp/language/template_argument_deduction .